### PR TITLE
Change and remove a required A level

### PR DESCRIPTION
**Pull request by tomas-stefano**
### Context

We need to add the ability for providers to change or remove a required A level they have added to TDA a course on Publish, in order to enable TDA applications.

### Changes proposed in this pull request

Screenshot 2024-07-01 at 10 08 23

### Guidance to review

1. Does the validations work?
2. Does the change work? (from other subject to any subject and vice versa)
3. Does the remove work? (when you remove any or all subjects)

To enable the TDA locally you need to run the rollover script and be on 2025 cycle (see rollover guide for more details)

### Trello card

https://trello.com/c/ic2phanj/1789-a-levels-4-7-add-change-and-remove-a-required-a-level-question-page-for-tda-courses-on-publish

### CODE REVIEW

**Comment by tomas-stefano** [comment]
[A levels 4/7 - Add change and remove a required A level question page for TDA courses on Publish](https://trello.com/c/ic2phanj/1789-a-levels-4-7-add-change-and-remove-a-required-a-level-question-page-for-tda-courses-on-publish)

**Review by CatalinVoineag — COMMENTED** [review]
This looks good to me. The only thing I found was the back-link when editing a subject. I thought it should go back to the list of subjects not the first step of adding a A level subject.


https://github.com/DFE-Digital/publish-teacher-training/assets/11318084/2cd2201f-9f56-4cc4-b20d-396e626f2a18

**Comment by tomas-stefano** [comment]
> This looks good to me. The only thing I found was the back-link when editing a subject. I thought it should go back to the list of subjects not the first step of adding a A level subject.

Good catch! I changed now :)

File: app/views/publish/courses/a_level_requirements/add_a_level_to_a_list/new.html.erb [hunk 1]
```diff
@@ -41,8 +41,20 @@
...
             <%= summary_list.with_row do |row| %>
               <%= row.with_key(classes: "govuk-!-font-weight-regular") { render ALevelSubjectRequirementRowComponent.new(subject) } %>
-              <%= row.with_action(text: "Change", href: "#") %>
-              <%= row.with_action(text: "Remove", href: "#") %>
+              <%= row.with_action(text: "Change", href:
+                    publish_provider_recruitment_cycle_course_a_levels_what_a_level_is_required_path(
+                      @provider.provider_code,
+                      @provider.recruitment_cycle_year,
+                      @course.course_code,
+                      uuid: subject["uuid"]
```
[the remark this answers is not in the document]
Comment (tomas-stefano) [reply app/views/publish/courses/a_level_requirements/add_a_level_to_a_list/new.html.erb]: Perfect. Done ♻️

**Review by CatalinVoineag — APPROVED** [review]

**Review by elceebee — APPROVED** [review]
Just one small comment

File: app/wizards/a_level_steps/remove_a_level_subject_confirmation.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    def next_step
+      if confirm_deletion? && no_a_level_subject_requirements?
+        :exit
+      else
+        :add_a_level_to_a_list
+      end
+    end
+
+    def confirm_deletion?
```
[the remark this answers is not in the document]
Comment (tomas-stefano) [reply app/wizards/a_level_steps/remove_a_level_subject_confirmation.rb]: Done!

**Pull request merged**